tests/PHPStan/Rules/Arrays/NonexistentOffsetInArrayDimFetchRuleTest.php

```diff
@@ -754,4 +754,30 @@ public function testBug2634(): void
 		$this->analyse([__DIR__ . '/data/bug-2634.php'], []);
 	}
 
+	public function testBug11946(): void
+	{
+		$this->analyse([__DIR__ . '/data/bug-11946.php'], [
+			[
+				'Offset -1 does not exist on string.',
+				21,
+			],
+			[
+				'Offset -1 does not exist on numeric-string.',
+				22,
+			],
+			[
+				'Offset -1 does not exist on non-empty-string.',
+				23,
+			],
+			[
+				'Offset -1 does not exist on non-falsy-string.',
+				24,
+			],
+			[
+				'Offset -1 does not exist on string.',
```

Comment:
@VincentLanglet I wonder whether we intentionally don't get a `lowercase-string` in the error?

Reply:
Yes, lowercase string is not mentioned when it's not needed to not make message overcomplicated.